patches/net/minecraft/client/data/models/ModelProvider.java.patch

```diff
@@ -0,0 +1,140 @@
+-        new BlockModelGenerators(modelprovider$blockstategeneratorcollector, modelprovider$iteminfocollector, modelprovider$simplemodelcollector).run();
+-        new ItemModelGenerators(modelprovider$iteminfocollector, modelprovider$simplemodelcollector).run();
++        registerModels(new BlockModelGenerators(modelprovider$blockstategeneratorcollector, modelprovider$iteminfocollector, modelprovider$simplemodelcollector), new ItemModelGenerators(modelprovider$iteminfocollector, modelprovider$simplemodelcollector));
+         modelprovider$blockstategeneratorcollector.validate();
+         modelprovider$iteminfocollector.finalizeAndValidate();
+         return CompletableFuture.allOf(
+@@ -63,12 +_,22 @@
+ 
+     @Override
+     public String getName() {
+-        return "Model Definitions";
++        return "Model Definitions - " + modId;
+     }
+ 
+     @OnlyIn(Dist.CLIENT)
+     static class BlockStateGeneratorCollector implements Consumer<BlockStateGenerator> {
+         private final Map<Block, BlockStateGenerator> generators = new HashMap<>();
++        private final Supplier<Stream<? extends Holder<Block>>> knownBlocks;
++
++        public BlockStateGeneratorCollector(Supplier<Stream<? extends Holder<Block>>> knownBlocks) {
++            this.knownBlocks = knownBlocks;
++        }
++
++        @Deprecated // Neo: Provided for vanilla/multi-loader compatibility, use above constructor
++        public BlockStateGeneratorCollector() {
++            this(BuiltInRegistries.BLOCK::listElements);
++        }
+ 
+         public void accept(BlockStateGenerator p_388748_) {
+             Block block = p_388748_.getBlock();
+@@ -79,9 +_,9 @@
+         }
+ 
+         public void validate() {
+-            Stream<Holder.Reference<Block>> stream = BuiltInRegistries.BLOCK.listElements().filter(p_388333_ -> true);
++            Stream<? extends Holder<Block>> stream = knownBlocks.get();
+             List<ResourceLocation> list = stream.filter(p_386843_ -> !this.generators.containsKey(p_386843_.value()))
+-                .map(p_386823_ -> p_386823_.key().location())
++                .map(p_386823_ -> p_386823_.unwrapKey().orElseThrow().location())
+                 .toList();
+             if (!list.isEmpty()) {
+                 throw new IllegalStateException("Missing blockstate definitions for: " + list);
+@@ -97,6 +_,16 @@
+     static class ItemInfoCollector implements ItemModelOutput {
+         private final Map<Item, ClientItem> itemInfos = new HashMap<>();
+         private final Map<Item, Item> copies = new HashMap<>();
++        private final Supplier<Stream<? extends Holder<Item>>> knownItems;
++
++        public ItemInfoCollector(Supplier<Stream<? extends Holder<Item>>> knownItems) {
++            this.knownItems = knownItems;
++        }
++
++        @Deprecated // Neo: Provided for vanilla/multi-loader compatibility, use above constructor
```

Comment:
Reword the comment to avoid using position, and instead refer to the additional parameter.
```suggestion
+        @Deprecated // Neo: Provided for vanilla/multi-loader compatibility. Use constructor with Supplier parameter.
```